src/calculation/CompositeReportBuilder.ts

```diff
@@ -22,13 +22,32 @@ export class CompositeReportBuilder<T extends PopulationGroupResult> extends Abs
   report: CompositeMeasureReport;
   compositeScoringType: CompositeScoreType;
   compositeFraction: { numerator: number; denominator: number };
+  weightedComponents: Record<string, number>;
 
   constructor(compositeMeasure: fhir4.Measure, options: CalculationOptions) {
     super(compositeMeasure, options);
 
     this.compositeScoringType = getCompositeScoringFromMeasure(compositeMeasure) ?? 'all-or-nothing';
     this.compositeFraction = { numerator: 0, denominator: 0 };
     this.options = options;
+    this.weightedComponents = {};
+
+    // if the weight is not specified, then we assume the weight is 1
+    compositeMeasure.relatedArtifact?.forEach(ra => {
+      if (ra.resource && ra.type === 'composed-of') {
+        let weight = 1;
+        ra.extension?.forEach(extension => {
+          if (
+            extension.url === 'http://hl7.org/fhir/us/cqfmeasures/StructureDefinition/cqfm-weight' &&
+            extension.valueDecimal
+          ) {
+            weight = extension.valueDecimal;
+          }
+        });
+        this.weightedComponents[ra.resource] = weight;
+      }
+    });
```

Comment:
Not an official reviewer, but wanted to share something I noticed when reading through the [CQFM Composite Measure profile ](https://build.fhir.org/ig/HL7/cqf-measures/StructureDefinition-composite-measure-cqfm.html)for my task: looks like the weight extension can be defined on the `relatedArtifact` entries, which we do check for here, but the extension could also be defined on `group.component` of the composite measure. So we might also want to do a check to see if `group` is defined on the composite and if it contains the weight extension as well

Reply:
![Screenshot 2023-05-17 at 2 58 47 PM](https://github.com/projecttacoma/fqm-execution/assets/87094479/12a28b59-9085-42c2-87ac-963525843990)

Reply:
Update - out of scope for this PR since we would need to add a check to our component extraction to also extract components from `group`.

Made [TACOSTRAT-1306](https://jira.mitre.org/browse/TACOSTRAT-1306) to address this